specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2024-04-02-preview/managedClusters.json

```diff
@@ -6948,6 +6951,9 @@
           "x-ms-identifiers": [],
           "description": "List of orchestrator types and versions available for upgrade."
         },
+        "componentsByReleases": {
```

Comment:
I hadn't realized that after I approved this, the name of the property was changed from componentVersions to componentsByReleases. I'm not convinced this is a better name (though I do like the structure that Haitao proposed).

I would've suggested we call it componentsByVersion or maybe just components. The word release is confusing IMO.

Reply:
It's likely too late to fix this for this API version though. Couple options here:
1. Release as-is, change in the next version. It's a breaking change but that's allowed in preview so not a big deal.
2. Delay this feature to the next API version and fix it there.
3. Don't fix it at all and just go with `componentsByReleases`.

Reply:
@HuShaoRu any update?